### PR TITLE
feat: try to unwrap external errors during error wrapping

### LINKED ISSUE #37: Unpack external errors (especially coming from pkg/errors)
**Is your feature request related to a problem? Please describe.**
When getting errors from other libraries that use `github.com/pkg/errors` or forks of it, I noticed that the error is just flattened by calling `error.Error()` and then returning a `rootError`. 
This is fine for custom errors, but `pkg/errors` is very widespread and integration with it would make the created errors more complete.

**Describe the solution you'd like**
It would be nice if `eris` tries to unpack the error instead of just turning it into string, by recursively calling `Unpack`. As `pkg/errors` is very popular, checking for the old `func Cause() error` in addition to `func Unpack() error` would be great.

Not just errors from `pkg/errors` but any error in general that implements `Unpack` makes sense to unpack in my opinion.

Not sure how to unmangle the stack trace though.

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by morningvera**
closes #37

### CODE REVIEW

File: eris.go [hunk 1]
```diff
@@ -72,10 +74,35 @@ func wrap(err error, msg string) error {
 			root.stack.insertPC(*stack)
 		}
 	default:
+		// attempt to unwrap external errors while building a new error chain or fallback to flattening them
+		var errStr []string
+		for e != nil {
+			str := e.Error()
+			errStr = append([]string{str}, errStr...)
+			e = Unwrap(e)
```
Comment (sum2000) [inline eris.go]: Instead of assuming other libs, is it possible to check for pkg errors by using reflect type?

File: eris.go [hunk 2]
```diff
@@ -72,10 +74,35 @@ func wrap(err error, msg string) error {
...
+			if e != nil && e.Error() == str {
+				e = Unwrap(e)
+			}
+		}
 		err = &rootError{
-			msg:   e.Error(),
+			msg:   errStr[0],
 			stack: stack,
 		}
+		for i := 1; i < len(errStr); i++ {
```
Comment (sum2000) [inline eris.go]: we can potentially put this parsing in condition if type of err is pkg/error type

File: eris_test.go [hunk 3]
```diff
@@ -85,6 +76,104 @@ func TestErrorWrapping(t *testing.T) {
 	}
 }
 
+type withMessage struct {
+	msg string
+}
+
+func (e *withMessage) Error() string { return e.msg }
```
Comment (sum2000) [inline eris_test.go]: LOL

**Comment by sum2000** [comment]
I am on the fence about this approach, it feels forced and hard to manage, makes me think if we really should support it if this is the only way to do it

File: eris.go (hunk 1 above)
Comment (morningvera) [reply eris.go]: why wouldn't we want it to work for other packages too?

File: eris.go (hunk 2 above)
Comment (morningvera) [reply eris.go]: the parsing isn't just for pkg/errors types, it's for all external errors that can be unwrapped

**Pull request merged**